Extension/src/main.ts

```diff
@@ -104,43 +104,39 @@ export async function activate(context: vscode.ExtensionContext): Promise<CppToo
             // Reset the persistent boolean tracking whether to warn the user of architecture mismatch on OSX.
             promptForMacArchictureMismatch.Value = true;
             errMsg = localize("native.binaries.not.supported", "This {0} {1} version of the extension is incompatible with your OS. Please download and install the \"{2}\" version of the extension.", GetOSName(installedPlatformAndArchitecture.platform), installedPlatformAndArchitecture.architecture, vsixName);
-            vscode.window.showErrorMessage(errMsg, downloadLink).then(async (selection) => {
-                if (selection === downloadLink) {
-                    vscode.env.openExternal(vscode.Uri.parse(releaseDownloadUrl));
-                }
-            });
+            const selection: string | undefined = await vscode.window.showErrorMessage(errMsg, downloadLink);
+            if (selection === downloadLink) {
+                vscode.env.openExternal(vscode.Uri.parse(releaseDownloadUrl));
+            }
         }
     } else {
         // Reset the persistent boolean tracking whether to warn the user of architecture mismatch on OSX.
         promptForMacArchictureMismatch.Value = true;
         if (!(await util.checkInstallBinariesExist())) {
             errMsg = localize("extension.installation.failed", "The C/C++ extension failed to install successfully. You will need to repair or reinstall the extension for C/C++ language features to function properly.");
             const reload: string = localize("remove.extension", "Attempt to Repair");
-            vscode.window.showErrorMessage(errMsg, reload).then(async (value?: string) => {
-                if (value === reload) {
-                    await util.removeInstallLockFile();
-                    vscode.commands.executeCommand("workbench.action.reloadWindow");
-                }
-            });
+            const selection: string | undefined = await vscode.window.showErrorMessage(errMsg, reload);
+            if (selection === reload) {
+                await util.removeInstallLockFile();
+                vscode.commands.executeCommand("workbench.action.reloadWindow");
+            }
         } else if (!(await util.checkInstallJsonsExist())) {
             // Check the Json files to declare if the extension has been installed successfully.
             errMsg = localize("jason.files.missing", "The C/C++ extension failed to install successfully. You will need to reinstall the extension for C/C++ language features to function properly.");
             const downloadLink: string = localize("download.button", "Go to Download Page");
-            vscode.window.showErrorMessage(errMsg, downloadLink).then(async (selection) => {
-                if (selection === downloadLink) {
-                    vscode.env.openExternal(vscode.Uri.parse(releaseDownloadUrl));
-                }
-            });
+            const selection: string | undefined = await vscode.window.showErrorMessage(errMsg, downloadLink);
```

Comment:
This looks kinda dangerous.  The previous behavior was to show the dialog, and IF someone clicked on the URL, to open it.  The new behavior is to pause async execution of the extension UNTIL someone clicks on the URL (or dismisses the dialog?).  I think this should be restored to the previous approach.  You might want to check to make sure you're not waiting for the results of dialogs elsewhere, in similar scenarios.

Reply:
this one is resolved. I will check all other places before closing this comment.